### PR TITLE
Ship separate ES5/ES2015+ bundles

### PR DESCRIPTION
**Pull request by hborghols**
### This PR:

Adds support to ship separate ES5 (as `[name]index.js`) and ES2015+ (`as [name]modern.js`) bundles. This allows us to only ship ES2015+ code to users with modern browsers and thus reduce the required data. This PR adds build, compile and bundle scripts in the `scripts` folder. The bundle script contains the webpack configs for the respective bundles.  

proposed release: `major`

**Checklist** 🛡
- [x] I have exported my addition from `src/index.ts` (check if not applicable)
- [x] Appropriate tests have been added for my functionality (check if not applicable)
- [x] A designer has seen and approved my changes (check if not applicable)
- [ ] I have tested my addition in all supported browsers and for responsiveness (Chrome, Firefox, Safari, Edge, IE11 and mobile browsers)

**Adds** ✨
- Support for shipping modern and legacy JS bundles.

**Changes** 🌀
- The build config

**Removes** 👋
- `webpack/webpack.config.js` and replaces it with `scripts/bundles.js`.

### CODE REVIEW

File: scripts/bundles.js [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        new webpack.DefinePlugin({
+            'process.env.NODE_ENV': JSON.stringify('production'),
+        }),
+        new PeerDepsExternalsPlugin(),
+    ],
+};
+
+const modernConfig = Object.assign({}, baseConfig, {
+    output: {
+        filename: '[name]index.es2015.js',
```
Comment (woutsluiter) [inline scripts/bundles.js]:
```suggestion
        filename: '[name]modern.js',
```

Comment (woutsluiter) [reply scripts/bundles.js]: Gives a nicer import imo: `import Button from '@myonlinestore/bricks/lib/components/Button/modern';`

File: package.json [hunk 2]
```diff
@@ -90,13 +90,13 @@
...
     "test": "jest --config=./config/jest/jest.config.json",
     "test-coverage": "jest --config=./config/jest/jest.config.json --coverage --runInBand --bail",
-    "build": "rimraf ./lib && webpack --config ./config/webpack/webpack.config.js",
     "lint-fix": "tslint --fix --project ./config/typescript/tsconfig.json --config ./config/typescript/tslint.json ./src/**/*.{ts,tsx}",
     "lint": "tslint --project ./config/typescript/tsconfig.json --config ./config/typescript/tslint.json ./src/**/*.{ts,tsx}",
     "watch": "start-storybook -p 9001 -c ./config/storybook",
     "deploy-storybook": "storybook-to-ghpages",
     "build-now": "build-storybook -c ./config/storybook -o /public",
     "build-storybook": "build-storybook -c ./config/storybook -o .out",
+    "build": "rimraf ./lib && node scripts/build.js",
```
Comment (woutsluiter) [inline package.json]: Not specific for this line but could you change the `files` key from `dist` to `lib`?

File: scripts/bundles.js [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+            sourceMap: true,
+        }),
+        new webpack.DefinePlugin({
+            'process.env.NODE_ENV': JSON.stringify('production'),
+        }),
+        new PeerDepsExternalsPlugin(),
+    ],
+};
+
+const modernConfig = Object.assign({}, baseConfig, {
```
Comment (woutsluiter) [inline scripts/bundles.js]: The config isn't deep merged, so the plugins are overwritten. Maybe you could use https://github.com/survivejs/webpack-merge for this

File: scripts/bundles.js [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+            analyzerMode: 'static',
+            openAnalyzer: false,
+            reportFilename: '../reports/webpack/statistics-tree-modern.html',
+        }),
+    ],
+});
+
+const legacyConfig = Object.assign({}, baseConfig, {
+    output: {
+        filename: '[name]index.es5.js',
```
Comment (woutsluiter) [inline scripts/bundles.js]: And this could be `index.js`

**Comment by hborghols** [comment]
It seems like the modern and legacy bundles are nearly identical for some reason. This is iceboxed until we've had time for future research.

**Pull request closed without merging**